cql3/query_processor.cc

```diff
@@ -693,7 +693,7 @@ query_processor::prepare(sstring query_string, const service::client_state& clie
                                 std::numeric_limits<uint16_t>::max()));
                 }
                 SCYLLA_ASSERT(bound_terms == prepared->bound_names.size());
-                return make_ready_future<std::unique_ptr<statements::prepared_statement>>(std::move(prepared));
+                co_return co_await make_ready_future<std::unique_ptr<statements::prepared_statement>>(std::move(prepared));
```

Comment:
just `co_return co_await std::move(prepared); ` -  should compile in this case, no?

maybe even move is not needed (copy elision) - although it's a bit blurry to me if it's mandatory for `co_return` the same as for `return`

Reply:
Of course, if you want to return a value x from a coroutine, "co_return x;" is the right way to do it, and replaces the "return make_ready_future<...>(x)" in old code.